File: code-samples/ai/code-snippets/check-task.java
package ReadAsyncTask;

import java.io.IOException;
import com.google.common.reflect.TypeToken;
import com.google.gson.Gson;

import com.ringcentral.*;
import com.ringcentral.definitions.*;

public class ReadAsyncTask {
    static RestClient restClient;

    public static void main(String[] args) {
      var obj = new ReadAsyncTask();
      try {
        // Instantiate the SDK
        restClient = new RestClient("PRODUCTION-APP-CLIENT-ID", "PRODUCTION-APP-CLIENT-SECRET", "https://platform.ringcentral.com");

        // Authenticate a user using a personal JWT token
        restClient.authorize("PRODUCTION-JWT");

        obj.check_task_status("JOBID");

      } catch (RestException e) {
        System.out.println(e.getMessage());
      } catch (IOException e) {
        e.printStackTrace();
      }
    }
    /*
    * Check task status
    */
    private void check_task_status(String jobId)
    {
      try {
        var resp = restClient.ai().status().v1().jobs(jobId).get();
        @SuppressWarnings("serial")
        String jsonStr = new Gson().toJson(resp, new TypeToken<Object>(){}.getType());
        System.out.println(jsonStr);
      } catch (Exception ex) {
        System.out.println("Unable to read async task status. " + ex.getMessage());
      }
    }
}

File: code-samples/ai/code-snippets/check-task.cs
using System;
using System.IO;
using System.Threading.Tasks;
using System.Collections.Generic;
using RingCentral;
using Newtonsoft.Json;

namespace ReadAsyncTask {
  class Program {
    static RestClient restClient;
    static async Task Main(string[] args){
      try
      {
        // Instantiate the SDK
        restClient = new RestClient("RC_APP_CLIENT_ID", "RC_APP_CLIENT_SECRET", "https://platform.ringcentral.com");

        // Authenticate a user using a personal JWT token
        await restClient.Authorize("RC_USER_JWT");

        await check_task_status("JOBID");
      }
      catch (Exception ex)
      {
        Console.WriteLine("Unable to authenticate to platform. Check credentials. " + ex.Message);
      }
    }
    /*
    * Check async task status
    */
    static private async Task check_task_status(String jobId)
    {
      try
      {
        var resp = await restClient.Ai().Status().V1().Jobs(jobId).Get();
        Console.WriteLine(JsonConvert.SerializeObject(resp));
      }
      catch (Exception ex)
      {
        Console.WriteLine("Unable to read async task status. " + ex.Message);
      }
    }
  }
}
